### ISSUE TITLE
zsh autocompletion fails when short options are combined

### ISSUE THREAD

**Issue by jfritzbarnes** [docker/cli#2625]
**Description**

Zsh cli autocompletion is broken when combining short options, e.g. `docker exec -it container-name bash`

**Steps to reproduce the issue:**
1. Run zsh shell with docker autocompletion enabled
2. Type `docker exec -it ` and press `Tab`

> This was initially reproduced through [oh-my-zsh](https://github.com/robbyrussell/oh-my-zsh) docker plugin.

**Describe the results you received:**

```
➜  music-learner git:(master) ✗ docker exec -it
!                                                dirs
-                                                dirstack
.                                                dis_aliases
.keepme                                          dis_functions
0                                                dis_galiases
1                                                dis_saliases
2                                                disable
2to3                                             disklabel
2to3-                                            disktool
2to3-2.7                                         diskutil
2to3-3.7                                         disown
3                                                dispqlen.d
4                                                distnoted
5                                                ditto
6                                                djpeg
```

Partial list. This seems to be listing all possible commands from paths plus some additional options; however, it is not listing valid containers that are running.

**Describe the results you expected:**

Results expected are the same as specifying the short options not combined together:

```
➜  music-learner git:(master) ✗ docker exec -i -t
17e4ff21100b         proj1.                      --      21 minutes, jfritzbarnes/container
8ab186098832         proj1_db_1                  --        28 hours, mysql:5.7
ddb19ab435b1         proj2_db_1                  --         3 weeks, mysql:5.7
```
Partial list, container names/images modified.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker info`:**

```
Client:
 Debug Mode: false

Server:
 Containers: 57
  Running: 9
  Paused: 0
  Stopped: 48
 Images: 516
 Server Version: 19.03.8
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.64GiB
 Name: docker-desktop
 ID: VN6H:NA52:FRS5:VMHM:GHJJ:6JXF:ODZ4:VP2Y:T4IF:5WBP:O2C3:BGHV
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 148
  Goroutines: 174
  System Time: 2020-07-10T20:16:13.0083239Z
  EventsListeners: 3
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by ratijas** [comment]
I remember being frustrated by this behavior for the first time, like maybe five years ago? And so many year later I still wonder, how many more we should suffer, until some brave hero finally shows up to defeat the bug...

By the way, here is a link to the [original issue](https://github.com/ohmyzsh/ohmyzsh/issues/9266). There are instructions in the plugin's source on how to enable _Short-option stacking_, which is really a hack rather than a proper solution.

tl;dr:

```zsh
zstyle ':completion:*:*:docker:*' option-stacking yes
zstyle ':completion:*:*:docker-*:*' option-stacking yes
```

**Comment by ratijas** [comment]
Duplicate of https://github.com/docker/cli/issues/993. Should be closed. @jfritzbarnes

**Comment by thaJeztah** [comment]
Yes, this looks to be a duplicate of https://github.com/docker/cli/issues/993

As mentioned above, you need to enable "short options stacking" make completion work for combinations of short options; see https://github.com/docker/cli/blob/64b2e952875788b0162b821dc4fe51e6947f9136/contrib/completion/zsh/_docker#L42-L49

The original intent was to enable this by default, but unfortunately, doing so caused various issues (see the discussion on https://github.com/moby/moby/pull/17124), so for that reason, it was disabled by default.

Let me close this ticket, as it's a duplicate of https://github.com/docker/cli/issues/993
